cpp/include/logistic_regression.h

```diff
@@ -0,0 +1,102 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2016 Northeastern University
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef CPP_INCLUDE_LOGISTIC_REGRESSION_H_
+#define CPP_INCLUDE_LOGISTIC_REGRESSION_H_
+
+#include <string>
+#include <iostream>
+#include <cmath>
+#include "include/matrix.h"
+#include "include/vector.h"
+#include "include/kernel_types.h"
+#include "Eigen/SVD"
+#include "include/svd_solver.h"
+#include "include/util.h"
+
+namespace Nice {
+
+// Abstract class of common logistic regression functions
+template<typename T>
+class LogisticRegression {
+ private:
+  T Sigmoid(T x){
```

Comment:
1. This is not really the sigmoid function as it is lacking e.
2. I do not see any calls of Sigmoid, you probably want to call it inside your hypothesis function h()
3. If you want to just do everything in h(), it is fine, but please remove the unused function.